packages/dataviews/src/filter-and-sort-data-view.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * External dependencies
+ */
+import removeAccents from 'remove-accents';
+
+/**
+ * Internal dependencies
+ */
+import {
+	OPERATOR_IS,
+	OPERATOR_IS_NOT,
+	OPERATOR_IS_NONE,
+	OPERATOR_IS_ANY,
+	OPERATOR_IS_ALL,
+	OPERATOR_IS_NOT_ALL,
+} from './constants';
+import { normalizeFields } from './normalize-fields';
+
+function normalizeSearchInput( input = '' ) {
+	return removeAccents( input.trim().toLowerCase() );
+}
+
+const EMPTY_ARRAY = [];
+
+/**
+ * Applies the filtering, sorting and pagination to the raw data based on the view configuration.
+ *
+ * @param {any[]}    data   Raw data.
+ * @param {Object}   view   View config.
+ * @param {Object[]} fields Fields config.
+ *
+ * @return {Object} { data: any[], paginationInfo: { totalItems: number, totalPages: number } }
+ */
+export function filterAndSortDataView( data, view, fields ) {
+	if ( ! data ) {
+		return {
+			data: EMPTY_ARRAY,
+			paginationInfo: { totalItems: 0, totalPages: 0 },
+		};
+	}
+	const _fields = normalizeFields( fields );
+	let filteredData = [ ...data ];
+	// Handle global search.
+	if ( view.search ) {
+		const normalizedSearch = normalizeSearchInput( view.search );
+		filteredData = filteredData.filter( ( item ) => {
+			return _fields
+				.filter( ( field ) => field.enableGlobalSearch )
+				.map( ( field ) => {
+					return normalizeSearchInput( field.getValue( { item } ) );
+				} )
+				.some( ( field ) => field.includes( normalizedSearch ) );
+		} );
+	}
+
+	if ( view.filters.length > 0 ) {
+		view.filters.forEach( ( filter ) => {
+			const field = _fields.find(
+				( _field ) => _field.id === filter.field
+			);
+			if (
+				filter.operator === OPERATOR_IS_ANY &&
+				filter?.value?.length > 0
+			) {
+				filteredData = filteredData.filter( ( item ) => {
+					return filter.value.includes( field.getValue( { item } ) );
```

Comment:
This works for the fields in use by now (fields that are strings), though we'll need to support fields that are arrays.

Reply:
I can update and add a test using the new `categories` field.

Reply:
For the record my update that got lost regarding this part was:
```
if ( view.filters.length > 0 ) {
		view.filters.forEach( ( filter ) => {
			if ( ! filter?.value || ! filter?.value?.length ) {
				return;
			}
			const field = _fields.find(
				( _field ) => _field.id === filter.field
			);
			const filterFn = {
				[ OPERATOR_IS_ANY ]: ( item ) => {
					return filter.value.includes( field.getValue( { item } ) );
				},
				[ OPERATOR_IS_NONE ]: ( item ) => {
					return ! filter.value.includes(
						field.getValue( { item } )
					);
				},
				[ OPERATOR_IS ]: ( item ) =>
					filter.value === field.getValue( { item } ),
				[ OPERATOR_IS_NOT ]: ( item ) =>
					filter.value !== field.getValue( { item } ),
			};
			filteredData = filteredData.filter( filterFn[ filter.operator ] );
		} );
	}
```